core/src/main/java/org/infinispan/remoting/transport/jgroups/JGroupsTransport.java

```diff
@@ -146,7 +146,7 @@
  */
 @Scope(Scopes.GLOBAL)
 @JGroupsProtocolComponent("JGroupsMetricsMetadata")
-public class JGroupsTransport implements Transport, ChannelListener, AddressGenerator {
+public class JGroupsTransport implements Transport {
```

Comment:
The new classes are unnecessary and bring no benefit (except increasing the memory usage).

Reply:
The intention was to limit the public methods (and thus API) of that class which is a generally recommended thing to do (encapsulation, hiding implementation details etc).
Please confirm if you still don't agree and I will revert.

Reply:
I don't mind those methods being public. Please revert it. Thanks.

Reply:
Similar thing is done with the ChannelCallbacks class (previously). Should I also remove that one? Or leave it? Or maybe merge all internal classes to one?

Reply:
Merging everything in `ChannelCallbacks` is fine.